### ISSUE TITLE
[b2g][unverified] fatal error when trying to reuse unverified email address 

### ISSUE THREAD

**Issue by jrgm** [mozilla/persona#2902]
STR

1) Using a current unagi build on a phone, configure /data/local/user.js to enable native navigator.id:

```
pref("dom.identity.enabled", true);
pref("toolkit.identity.debug", true);
pref("layout.css.report_errors", false);
```

(The layout.css.report_errors isn't necessary, but seriously, the noise about CSS "declaration dropped" is useless information).
2) Use the firefox browser on the phone and go to `https://marketplace-dev.allizom.org`
3) Click on `Log in/Register` to bring up the trusted UI.
4) Enter an email address that has not been used before; then enter a password twice. You will be authenticated.
5) (Also an email to confirm that address will be sent out; do not verify that email address).
6) Now logout of marketplace-dev.
7) Click `Log in/Register` to bring up the dialog again.
8) Click `This is not me`. Re-enter the same email address from step 4) and a password twice.

Expected: this unverified user should be restaged again

Actual: "We are very sorry. The server is under extreme load."

```
Action: Creating Account
Network Info: POST /wsapi/stage_user
Response Code: 503
Response Text/Error Type: Service Unavailable: database unavailable
```

In the dbwriter.log on notoriousb2g.personatest.org:

```
unexpected database failure: Error: Duplicate entry 'blahblah@gmail.com' for key 'address' -- at exports.createUnverifiedUser
(/home/app/code/lib/db/mysql.js:680:13)
```

**Comment by lloyd** [comment]
@ozten - are you the right guy to take a look at this?

**Comment by seanmonstar** [comment]
No, I am. _claims_

**Comment by jds2501** [comment]
Note - this is blocking gaia ui test automation & a smoke test on device. So some of the QA dept is blocked by this bug right now.

**Comment by bobsilverberg** [comment]
The sequence of events I am seeing is slightly different, in the gaia-ui-test for test_marketplace_login. I am not using the prefs stated above, and the sequence is as follows:
1. Launch the Marketplace app on the phone.
2. Click the Login button at the bottom of the page.
3. "Firefox Accounts powered by Persona" screen appears, with "To sign in with Persona, please enter your email address."
4. Enter a valid email address and click next.
5. A Persona screen appears with two password fields, "create password" and "verify password".
6. Enter the valid password into each field and click done.
7. The screen with "We are very sorry. The server is under extreme load!" appears.

**Comment by seanmonstar** [comment]
This should be fixed on notoriousb2g.personatest.org

**Comment by jrgm** [comment]
This works for me on notoriousb2g.personatest.org for the original STR. @bobsilverberg do you still have an issue, after 2013-01-04T20:02:41.145Z?

**Comment by bobsilverberg** [comment]
It is working for me now too. I am still having a problem with the actual sign in. After the Persona sign in is completed and I am returned to the Marketplace app the app never signs me in. It just sits with the "Log in / Register" button at the bottom and a spinner to the right of that.

Should I open a new issue for that here, or is that something I should add to Bugzilla, and if so, under what component?

**Comment by jds2501** [comment]
@bobsilverberg I can reproduce the exact STR you just specified with window.open. Let me dig into this.

**Comment by jds2501** [comment]
01-04 20:51:17.136: E/GeckoConsole(558): [JavaScript Error: "Error: Syntax error, unrecognized expression: Persona authentication ." {file: "https://marketplace.cdn.mozilla.net/media/js/mkt/consumer-min.js?build=99103e3" line: 12}]

**Comment by jds2501** [comment]
My general observer test case worked. I'm guessing there's a marketplace bug here.

**Comment by jds2501** [comment]
@bobsilverberg Sounds like you've hit https://bugzilla.mozilla.org/show_bug.cgi?id=825306
